lua/neogit/popups/tag/actions.lua

```diff
@@ -0,0 +1,141 @@
+local M = {}
+local git = require("neogit.lib.git")
+local client = require("neogit.client")
+local utils = require("neogit.lib.util")
+local FuzzyFinderBuffer = require("neogit.buffers.fuzzy_finder")
+local input = require("neogit.lib.input")
+
+function M.create_tag(popup)
+  local tag_input = input.get_user_input("Tag name: ")
+  if not tag_input or tag_input == "" then
+    return
+  end
+  local options = git.refs.get_revisions()
+  local selected_branch = FuzzyFinderBuffer.new(options):open_async()
+  if not selected_branch then
+    return
+  end
+  local args = popup:get_arguments()
+  if vim.tbl_count(args) > 0 and not vim.tbl_contains(args, "--annotate") then
+    table.insert(args, "--annotate")
+  end
+
+  client.wrap(git.cli.tag.arg_list(utils.merge(args, { tag_input, selected_branch })), {
+    autocmd = "NeogitTagComplete",
+    msg = {
+      success = "Added tag " .. tag_input .. " on " .. selected_branch,
+      fail = "Failed to add tag " .. tag_input .. " on " .. selected_branch,
+    },
+  })
+end
+
+--- Create a release tag for `HEAD'.
+---@param _ table
+function M.create_release(_) end
+
+--- Delete one or more tags.
+--- If there are multiple tags then offer to delete those.
+--- Otherwise prompt for a single tag to be deleted.
+--- git tag -d TAGS
+---@param _ table
+function M.delete(_)
+  local a = require("plenary.async")
+  local notif = require("neogit.lib.notification")
+
+  local options = git.tag.list()
+  local tags = {}
+  if options ~= nil then
+    local selected_tags = FuzzyFinderBuffer.new(options):open_async { allow_multi = true }
+    if #selected_tags == 0 then
+      return
+    end
+    tags = selected_tags
+  else
+    local tag_input = input.get_user_input("Tag name:")
+    if not tag_input or tag_input == "" then
+      return
+    end
+    table.insert(tags, tag_input)
```

Comment:
I think we can remove this else clause. Two reasons: 
- If `git.tag.list()` doesn't find it.. is there anything to delete?
- The fuzzy finder buffer works as an input when no entries match the prompt. I just added that yesterday. So a user can enter whatever they like, and if there's a prompt but no matches, we use the prompt input.

Reply:
Ah that makes sense!